include/blmc_robots/n_joint_blmc_robot_driver.hpp

```diff
@@ -458,4 +632,30 @@ class NJointBlmcRobotDriver
     }
 };
 
+template <typename Driver>
+typename Driver::Types::BackendPtr create_backend(
+    typename Driver::Types::DataPtr robot_data,
+    const std::string &config_file_path = "")
+{
+    constexpr double MAX_ACTION_DURATION_S = 0.003;
+    constexpr double MAX_INTER_ACTION_DURATION_S = 0.005;
+
+    std::cout << "Load robot driver configuration from file '"
+              << config_file_path << "'" << std::endl;
+    auto config = Driver::Config::load_config(config_file_path,
+                                              Driver::get_default_config());
+    config.print();
+
+    std::shared_ptr<
+        robot_interfaces::RobotDriver<typename Driver::Types::Action,
+                                      typename Driver::Types::Observation>>
+        robot = std::make_shared<Driver>(config);
```

Comment:
just for simplicity, couldn't we just use shared_ptr<Driver> as type on the left hand side? i think it would automatically be upcast below.

Reply:
Just using `std::shard_ptr` results in errors but `auto` is working.